src/core/border_router/routing_manager.cpp

```diff
@@ -313,7 +313,7 @@ void RoutingManager::HandleNotifierEvents(Events aEvents)
         EvaluateState();
     }
 
-    if (aEvents.Contains(kEventThreadNetdataChanged))
+    if (mIsRunning && aEvents.Contains(kEventThreadNetdataChanged))
```

Comment:
There is no problem with current implementation because we have `VerifyOrExit(mIsRunning);` in `EvaluateRoutingPolicy(...)`.